### PR TITLE
Add create reservation feature

### PR DESCRIPTION
**Pull request by maciejfedor**
I've added feature that allows to book a reservation for a chosen screening. If a seat is taken then checkbox for that seat is disabled.

- add create action to reservations controller (create reservation and its tickets)
- add method that checks available seats for a screening
- change seat selection form to checkboxes
- add error display when no seat is chosen
- minor fixes

### CODE REVIEW

**Review by h4ppyr0gu3 — APPROVED** [review]

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+
+  private
+
+def set_screening
+  @screening = Screening.find(params[:screening_id])
+end
+
+def create_tickets
+  params[:seats].each do |seat|
+    Ticket.create(reservation_id: @reservation.id, seat:)
```
Comment (agusiol) [inline app/controllers/reservations_controller.rb:31]:
```suggestion
    Ticket.create(reservation_id: @reservation.id, seat: seat)
```

File: app/models/hall.rb [hunk 2]
```diff
@@ -5,7 +5,8 @@ class Hall < ApplicationRecord
   validates :name, :capacity, presence: true
   validates :name, uniqueness: true
 
-  def seats_array
-    (1..capacity).to_a
+  def seats
+    arr = (1..capacity).to_a
+    arr.map(&:to_s)
```
Comment (agusiol) [inline app/models/hall.rb:10]: No need to set additional variable `arr` here.
`(1..capacity).map(&:to_s)` works as well

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (maciejfedor) [reply app/controllers/reservations_controller.rb:31]: I also thought it is wrong but that's how rubocop keeps correcting it so i left it that way. 
`app/controllers/reservations_controller.rb:31:58: C: [Corrected] Style/HashSyntax: Omit the hash value.
    Ticket.create(reservation_id: @reservation.id, seat: seat)`. 
    Should i exlcude HashSyntax in the rubocop config ?

Comment (agusiol) [reply app/controllers/reservations_controller.rb:31]: My mistake ;) . This is a new shorthand syntax in Hash literals in Ruby 3.1 https://blog.saeloun.com/2021/09/28/ruby-allow-value-omission-in-hash-literals.html

**Pull request merged**